lib/src/provider.dart

```diff
@@ -58,7 +58,12 @@ class Provider<T> extends InheritedWidget implements ProviderBase {
         ? context.inheritFromWidgetOfExactType(type) as Provider<T>
         : context.ancestorInheritedElementForWidgetOfExactType(type)?.widget
             as Provider<T>;
-    return provider?.value;
+
+    if (provider == null) {
```

Comment:
Should this be an assert?

This would fasten a bit the release build. That's what Flutter does with things like `Scaffold.of`